Newsletter2Go/Export/Block/Head.php

```diff
@@ -0,0 +1,174 @@
+            $result .= 'n2g("ecommerce:addItem", ' . $item . ');';
+        }
+
+        return $result;
+    }
+
+    /**
+     * Returns array of order data that based on given order id
+     *
+     * @param int $orderId
+     * @return array
+     */
+    private function getOrderData($orderId)
+    {
+        /** @var SalesModel\Order $order */
+        $order = $this->objectManager->get(SalesModel\Order::class)->load($orderId);
+        $storeName = explode(PHP_EOL, $order->getStoreName());
+        $result[] = [
+            'id' => $orderId,
+            'affiliation' => $storeName[0],
+            'revenue' => strval(round($order->getBaseGrandTotal(), 2)),
+            'shipping' => strval(round($order->getShippingAmount(), 2)),
+            'tax' => strval(round($order->getTaxAmount(), 2)),
+        ];
+
+        return $result;
+    }
+
+    /**
+     * Returns array of products based on given order id
+     *
+     * @param integer $orderId
+     * @return array
+     */
+    private function getItemData($orderId)
+    {
+        /** @var SalesModel\Order $order */
+        $order = $this->objectManager->get(SalesModel\Order::class)->load($orderId);
+        $result = [];
+
+        /** @var SalesModel\Order\Item $item */
+        foreach ($order->getAllVisibleItems() as $item) {
+            $itemId = $this->getParentProductId($item->getProduct());
+            $categoryName = $this->getCategoryName($itemId);
+            $result[] = [
+                'id' => $orderId,
+                'name' => $item->getName(),
+                'sku' => $item->getSku(),
+                'category' => $categoryName,
+                'price' => strval(round($item->getBasePrice(), 2)),
+                'quantity' => strval(round($item->getQtyOrdered(), 2)),
+            ];
+        }
+
+        return $result;
+    }
+
+    /**
+     * Returns category name based on given product id
+     *
+     * @param integer $itemId
+     * @return string
+     */
+    private function getCategoryName($itemId)
+    {
+        $result = '';
+        $product = $this->productRepository->getById($itemId);
+        $category = $product->getCustomAttribute('category_ids');
+        $categoryIds = $category->getValue();
+        if (count($categoryIds)) {
+            $category = $this->categoryRepository->get($categoryIds[0]);
+            $result = $category->getName();
+        }
+
+        return $result;
+    }
+
+    /**
+     * Retrieves parent product id if product is configurable
+     * @param CatalogModel\Product $product
+     * @return mixed
+     */
+    private function getParentProductId($product)
+    {
+        $id = $product->getId();
+        $type = $product->getTypeId();
+        if ($type == 'configurable') {
+            $parents = $this->objectManager
+                ->create(Configurable::class)
+                ->getParentIdsByChild($id);
+
+            if (!empty($parents)) {
+                return $parents[0];
```

Comment:
`array_shift()` is probably a better option; just in case the first item happened to have a different index.